src/main/java/gift/domain/product/repository/JpaProductRepository.java

```diff
@@ -0,0 +1,23 @@
+package gift.domain.product.repository;
+
+import gift.domain.product.Product;
+import java.util.List;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+@Repository
+public interface JpaProductRepository extends JpaRepository<Product, Long> {
+
+    boolean existsByName(String name);
+
+    @Transactional
+    @Modifying
+    @Query("DELETE FROM Product p WHERE p.id IN :productIds")
+    void deleteByIds(@Param("productIds") List<Long> productIds);
```

Comment:
요렇게 하시면 굳이 @Query를 사용하지 않아도 될 것 같습니당ㅎㅎ
```suggestion
    void deleteAllByIdsIn(List<Long> productIds);
```

Reply:
JPA 쿼리 메서드 중에 In 은 몰랐는데, 정말 다양한 기능을 지원하네요.. 감사합니다!
*찾아보니 deleteAllByIdIn 이 맞는 것 같습니다 ㅎㅎ